pkgs/fc/agent/fc/manage/collect_garbage.py

```diff
@@ -0,0 +1,136 @@
+import datetime
+import os
+import pwd
+import subprocess
+from pathlib import Path
+
+import structlog
+from typer import Exit, Option, Typer
+
+import fc.util.lock
+from fc.util.logging import init_logging
+
+app = Typer()
+
+
+@app.command()
+def collect_garbage(
+    verbose: bool = Option(
+        False, "--verbose", "-v", help="Show debug messages and code locations."
+    ),
+    exclude_file: Path = Option(
+        exists=True,
+        file_okay=True,
+        dir_okay=False,
+        readable=True,
+        default="/etc/fc-userscan.exclude",
+        help="File with exclude rules for fc-userscan",
+    ),
+    stamp_dir: Path = Option(
+        exists=True,
+        file_okay=False,
+        writable=True,
+        default="/var/log",
+        help="Location for the file with the last successful run timestamp.",
+    ),
+    lock_dir: Path = Option(
+        exists=True,
+        file_okay=False,
+        writable=True,
+        default="/run/lock",
+        help="Where the lock file for exclusive operations should be placed.",
+    ),
+):
+
+    init_logging(verbose, syslog_identifier="fc-collect-garbage")
+    log = structlog.get_logger()
+
+    log.debug("collect-garbage-start")
+
+    return_codes = []
+    users_to_scan = [
+        user
+        for user in pwd.getpwall()
+        if user.pw_uid >= 1000 and user.pw_dir != "/var/empty"
+    ]
+    log.info(
+        "userscan-start",
+        _replace_msg="Running fc-userscan for {user_count} users",
+        user_count=len(users_to_scan),
+    )
+
+    for user in users_to_scan:
+        log.debug(
+            "userscan-user",
+            _replace_msg="Scanning {homedir} as {name}",
+            homedir=user.pw_dir,
+            name=user.pw_name,
+        )
+
+        p = subprocess.Popen(
+            [
+                "fc-userscan",
+                "-r",
```

Comment:
Some of the short-form parameters here are non-obvious, it can improve the readability if these are commented on.